### PR TITLE
Rocket boots Rocket boots

### PR DESCRIPTION
**Pull request by YakumoChen**
## About The Pull Request
Adds Rocket Boots, a black market pair of jump boots that makes you slam into walls (or people) up to 20 tiles away

## Why It's Good For The Game
So people can give themselves brain damage in a different way than meth while they avoid the popo.

## Changelog
:cl:
add: Rocket boots, a bad idea from the black market that hurls you 20 tiles in the direction you're facing. Beware of walls and people.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
did you test this? pretty sure it will only be available to nukies

File: code/modules/clothing/shoes/miscellaneous.dm [hunk 1]
```diff
@@ -273,6 +273,13 @@
 	else
 		to_chat(user, "<span class='warning'>Something prevents you from dashing forward!</span>")
 
+/obj/item/clothing/shoes/bhop/rocket 
+	name = "rocket boots"
+	desc = "Very special boots with built-in rocket thrusters! SHAZBOT!"
+	var/jumpdistance = 20 //great for throwing yourself into walls and people at high speeds
+	var/jumpspeed = 5
+	var/recharging_rate = 60 //default 6 seconds between each dash
```
Comment (Fikou) [inline code/modules/clothing/shoes/miscellaneous.dm:281]: inherited from parent

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1602,6 +1602,14 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
 	cost = 2
 	illegal_tech = FALSE
+	
+/datum/uplink_item/device_tools/rocketboots
+	name = "Rocket Boots"
+	desc = "A pair of heavily modified jump boots with added rocket thrusters. Great for making quick escapes, although you need a very long, unobstructed space to not slam into anything."
+	item = /obj/item/clothing/shoes/bhop/rocket
+	include_modes = list(/datum/game_mode/nuclear, /datum/game_mode/nuclear/clown_ops)
+	cost = 3 //you have to choose between this or noslips. Personally I'd go with the noslips.
+	illegal_tech = FALSE //It's a bad idea, not an illegal one.
```
Comment (Fikou) [inline code/modules/uplink/uplink_items.dm:1612]: why though

**Comment by YakumoChen** [comment]
I didn't test this, this is a 6 AM bad idea. Hold on

File: code/modules/clothing/shoes/miscellaneous.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/clothing/shoes/miscellaneous.dm]: also try using 6 SECONDS instead of 60

**Comment by XDTM** [comment]
Even if it's a fun meme item, there are already complaints about the uplink being overbloated with items that are too niche to be useful most of the time

**Comment by Tlaltecuhtli** [comment]
if you dont know where to put it and want it to be a limited item, the blackmarket is a good alternative

**Comment by D4C-420** [comment]
seems like an impractical version of a miners jump boots. Outside of the long main halls (on everything but Kilo) I don't think any places are 20 tiles long. This would be better as some joke upgrade for the aforementioned boots maybe in some random ruin or vended from the mining vendor as some expensive item. Also go go gadget rocket boots.

File: code/modules/uplink/uplink_items.dm (hunk 2 above)
Comment (necromanceranne) [reply code/modules/uplink/uplink_items.dm:1611]: It's literally a traitor item, this should give you illegal tech.

**Comment by Learner10** [comment]
If it leaves a backblast, then we've got something spicy.

**Comment by YakumoChen** [comment]
> if you dont know where to put it and want it to be a limited item, the blackmarket is a good alternative

I haven't heard about the black market. I'll look into that and move it then.

@D4C-420 the intent was to make something similar obtainable without being a miner.

**Comment by D4C-420** [comment]
@YakumoChen Well I have no idea what the black market is but I'm guessing its not a tot item anymore making it suck slightly less (still like 70% of the times the area you are in is going to be too cramped to make use of them) but even then the miner's boots are kind of meh already.

**Comment by YakumoChen** [comment]
> @YakumoChen Well I have no idea what the black market is but I'm guessing its not a tot item anymore making it suck slightly less (still like 70% of the times the area you are in is going to be too cramped to make use of them) but even then the miner's boots are kind of meh already.

It's great for long hallways! I hold no warranty if you hit something 2 screens away you had no way of seeing. The long distance should make up for any knockdowns before you're caught up to. Probably. You can also boost over tables if you're dating enough.

**Comment by YakumoChen** [comment]
Done. It's now in the black market instead. I'm not sure if the price is fair or not, so I'm happy to take suggestions.

**Comment by Jared-Fogle** [comment]
> You are expected to have tested your pull requests if it is anything that would warrant testing. Text only changes, single number balance changes, and similar generally don't need testing, but anything else does. **This means by extension web edits are disallowed for larger changes.**

**Comment by YakumoChen** [comment]
@Jared-Fogle I would love to get off my ass and test it but I apparently _literally_ cannot due to how hard you made it to compile with as old as my computer is

**Comment by Jared-Fogle** [comment]
The best we can do is help you in #coding-general.

**Comment by YakumoChen** [comment]
I give up. I have tried everything to just so much as get a working compile on my computer I can edit so I can "test this" and I am unable to.

**Comment by YakumoChen** [comment]
alright well after trying everything short of installing a full git setup just to test my memey PR, nothing works, so I give up.

**Pull request closed without merging**